### PR TITLE
Install script and docker install

### PR DESCRIPTION
**Pull request by tristanlatr**
This PR include :
- A new install.sh script for quick install. 
- A Dockerfile to run wpscan-analyzer on a docker container
- A Readme update to explain new possible install ways

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+# This script will download, unzip and install binary file to /usr/local/bin/
+# Script requirement: wget, gzip
+# More infos https://github.com/lukaspustina/wpscan-analyze
+
+# Install config
+version=v1.0.4
```
Comment (lukaspustina) [inline install.sh]: We could last use master instead of a pinned version.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    fi
+fi
+# Download
+echo "Downloading wpscan-analyze ${version} x64 bits $(uname) binary"
+# Remove previous file if any
+rm "${filename}.gz" 2>/dev/null || true
+# Get binary file
+wget --quiet "https://github.com/lukaspustina/wpscan-analyze/releases/download/${version}/${filename}.gz"
+# Remove previous file if any
+rm "${filename}" 2>/dev/null || true
```
Comment (lukaspustina) [inline install.sh]: Please move this before wget -- let's keep cleanup code together.

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Unzip it
+gzip -d "${filename}.gz"
+# Remove previous file if any
+rm "${binary_file}" 2>/dev/null || true
```
Comment (lukaspustina) [inline install.sh]: see above.

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# Copy file
+echo "Copying binary to ${binary_file}"
+cp "${filename}" "${binary_file}"
+# Make the file executable
+chmod +x ${binary_file}
+echo "Installed to:"
```
Comment (lukaspustina) [inline install.sh]:
```bash
/bin/echo -n "Installed to '${install_to}': "
if ....
else
  echo "ok"
fi

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+if ! which wpscan-analyze; then
+    echo "Seems that ${install_to} is not in your PATH"
+    exit 1
+fi
+echo "Success! Restart your terminal to use 'wpscan-analyze' command"
```
Comment (lukaspustina) [inline install.sh]: This isn't necessary unless you're using zsh and expect autocompletion -- like I do.

**Review by lukaspustina — CHANGES_REQUESTED** [review]
I like this PR. I just have few minor change requests. Thanks for your participation.

**Comment by tristanlatr** [comment]
Hi, 

I've implemented your comments into the script.
It will clone the repo to fetch latest tag and download latest version always.
Cleaning is done before copying/creating files so it cannot be all together at the beginning but I tried to make it more clear.

File: install.sh [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
+
 # Quit if not a 64 bits arch
 if [ ! "$(uname -m)" = x86_64 ]; then
-    echo "Unsupported architecture, only 64 bits are supported by this script. Please install Rust environment and build wpscan-analayze from source. Check https://github.com/lukaspustina/wpscan-analyze for more infos."
+    echo "Unsupported architecture, only 64 bits are supported by this script. Please install Rust environment and build wpscan-analayze from source. Visit https://github.com/lukaspustina/wpscan-analyze for more infos."
     exit 1
 fi
+
+# Determine version based on lastest git tag
+git clone https://github.com/lukaspustina/wpscan-analyze --quiet
```
Comment (lukaspustina) [inline install.sh]: How about `curl --silent "https://api.github.com/repos/lukaspustina/wpscan-analyze/releases/latest" | grep tag_name | sed -E 's/.*"v(.*)",/\1/'`

File: install.sh [hunk 7]
```diff
@@ -26,30 +35,29 @@ else
...
+        echo "Unsupported operating system, please install Rust environment and build wpscan-analayze from source. Visit https://github.com/lukaspustina/wpscan-analyze for more infos."
         exit 1
     fi
 fi
 # Download
-echo "Downloading wpscan-analyze ${version} x64 bits $(uname) binary"
-# Remove previous file if any
+echo "Downloading wpscan-analyze ${version} $(uname -m) $(uname) binary"
+
+# CLeaning before downloading binary
```
Comment (lukaspustina) [inline install.sh]:
```suggestion
# Cleaning before downloading binary
```

File: install.sh [hunk 8]
```diff
@@ -26,30 +35,29 @@ else
...
+
 # Get binary file
 wget --quiet "https://github.com/lukaspustina/wpscan-analyze/releases/download/${version}/${filename}.gz"
-# Remove previous file if any
-rm "${filename}" 2>/dev/null || true
+
 # Unzip it
 gzip -d "${filename}.gz"
-# Remove previous file if any
-rm "${binary_file}" 2>/dev/null || true
```
Comment (lukaspustina) [inline install.sh]: Just FYI: `rm -f` always returns 0

**Review by lukaspustina — CHANGES_REQUESTED** [review]
I think, I've found a better way to get the latest release version. That's the only change I need.

File: install.sh (hunk 6 above)
Comment (tristanlatr) [reply install.sh]: Ok I'll use that !

File: install.sh (hunk 8 above)
Comment (tristanlatr) [reply install.sh]: I did not know !

**Review by lukaspustina — APPROVED** [review]

**Comment by lukaspustina** [comment]
LGTM. Thanks for your contribution.

**Comment by tristanlatr** [comment]
@lukaspustina 
There is an issue with the script, It's working when calling `./install.sh` but not when we do `cat install.sh | sh`.
So the curl command in the readme is not working :/

**Pull request merged**